### ISSUE TITLE
colVis + stateSave leads to broken filters

### ISSUE THREAD

**Issue by jonnsn** [vedmack/yadcf#265]
Hi,

I have a Datatable with colVis Button and stateSave option set to true.
Once I hide one or more columns and reload the page, the columns are still hidden therefore. If I want to show the columns now, there are no filters for them and I can't seem to add them on 'column-visibility.dt'.
I tried this for example:
```javascript
dataTable.on('column-visibility.dt', function (e, settings, col_num, state) {
		var obj = {};
		obj[col_num] = yadcf.getOptions(settings.oInstance.selector)[col_num];
		if (obj[col_num] !== undefined) {
			obj[col_num].column_number = col_num;
			if (obj[col_num] !== undefined) {
				appendFilters(settings.oInstance.selector,
					obj,
					settings.oInstance.selector);
			}
		}
	  });
```
but obviously appendFilters is not accessible here and I could not find another possibility to init the "new" columns.

I'd be happy for any suggestions :-)

**Comment by 1nickt** [comment]
I also have this behavior with colVis and stateSave.

**Comment by vedmack** [comment]
please provide a link to a jsbin/jsfiddle/ etc so I can reproduce this scenario...

**Comment by 1nickt** [comment]
@vedmack Thank you for your attention, here is a demo: http://live.datatables.net/kitawawo/2/edit?html,js,output

**Comment by vedmack** [comment]
@jonnsn @1nickt , both of you are using datatables with DOM source? because if you do so then it didn't work only because I wasn't listening to the colvis events in DOM source mode, I just added it on my dropbox file and the jsbin works fine, let me know if it works for both of you and I'll commit it.
See it in action http://live.datatables.net/kitawawo/5/edit

**Comment by 1nickt** [comment]
Hi, I won't get a chance to review tonight, but no, I am using JavaScript
data source.

-nick

**Comment by jonnsn** [comment]
@vedmack : I use DOM source and it seems to fix the issue for me. Thank you very much for your help!

**Comment by 1nickt** [comment]
Hi again Daniel,

Fixed!

( Using Javascript Data Source )

See http://live.datatables.net/kitawawo/8/edit

I also tested with my real site.

You *are* the man.

Thanks very much.

By the way, I tried many ways to get column filter boxes at the top of my
table with JS data in DataTables, including the ones on the official site
docs, and other more visible plugins, but yours was the first and only one
that I was able to get working. Thank you!

- nick
